frontends/concrete-python/tests/execution/test_examples.py

```diff
@@ -0,0 +1,248 @@
+
+    replace(db, 3, 1)
+    assert np.array_equal(
+        inspect(db),
+        [
+            [1] + [0, 0, 0, 3] + [0, 0, 0, 1],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+        ],
+    )
+
+    assert query(db, 3) == 1
+
+    insert(db, 25, 40)
+    assert np.array_equal(
+        inspect(db),
+        [
+            [1] + [0, 0, 0, 3] + [0, 0, 0, 1],
+            [1] + [0, 1, 2, 1] + [0, 2, 2, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+        ],
+    )
+
+    assert query(db, 25) == 40
+
+    minimum_key = 0
+    maximum_key = 2**db.key_size - 1
+
+    minimum_value = 0
+    maximum_value = 2**db.value_size - 1
+
+    db.initialize()
+    assert np.array_equal(
+        inspect(db),
+        [
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+        ],
+    )
+
+    insert(db, minimum_key, minimum_value)
+    assert np.array_equal(
+        inspect(db),
+        [
+            [1] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+        ],
+    )
+
+    assert query(db, minimum_key) == minimum_value
+
+    replace(db, minimum_key, maximum_value)
+    assert np.array_equal(
+        inspect(db),
+        [
+            [1] + [0, 0, 0, 0] + [3, 3, 3, 3],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+        ],
+    )
+
+    assert query(db, minimum_key) == maximum_value
+
+    insert(db, maximum_key, maximum_value)
+    assert np.array_equal(
+        inspect(db),
+        [
+            [1] + [0, 0, 0, 0] + [3, 3, 3, 3],
+            [1] + [3, 3, 3, 3] + [3, 3, 3, 3],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+        ],
+    )
+
+    assert query(db, maximum_key) == maximum_value
+
+    replace(db, maximum_key, minimum_value)
+    assert np.array_equal(
+        inspect(db),
+        [
+            [1] + [0, 0, 0, 0] + [3, 3, 3, 3],
+            [1] + [3, 3, 3, 3] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+            [0] + [0, 0, 0, 0] + [0, 0, 0, 0],
+        ],
+    )
+
+    assert query(db, maximum_key) == minimum_value
+
+
+@pytest.mark.parametrize(
+    "mode",
+    [
+        "simulate",
+        "fhe",
+    ],
+)
+def test_levenshtein_distance(mode, helpers):
+    """
+    Test levenshtein distance example.
+    """
+
+    configuration = helpers.configuration()
+    if mode == "simulate":
+        configuration = configuration.fork(fhe_execution=False, fhe_simulation=True)
+
+    alphabet = Alphabet.lowercase()
```

Comment:
- shouldn't we have tests with all alphabets?
- shouldn't we have a list of [alphabet, string1, string2, expected_distance] and just have a loop over that

Reply:
> shouldn't we have tests with all alphabets?

We can, but testing takes a lot of time. Just like we can't test every input for every function, it's okay to have few of the alphabets.

Reply:
> shouldn't we have a list of [alphabet, string1, string2, expected_distance] and just have a loop over that

We can, but the problem is recompilation. If we do it like that as pytest arguments, we would need to recompile the circuit for every input/output pair.

Reply:
- if we don't have a minimal test per alphabet, we can't be sure it works, in my opinion
- and for  [alphabet, string1, string2, expected_distance]:
        - if you think changing alphabet is too complicated, consider [ string1, string2, expected_distance] but my proposal would stand
        - I'm not speaking about having those list of [] in pyparameters, it could be in the function, but still, we would avoid to have too many assert lines

Reply:
> if we don't have a minimal test per alphabet, we can't be sure it works, in my opinion

Then what about the alphabets we don't have by name? How much of it we should test?

> I'm not speaking about having those list of [] in pyparameters, it could be in the function, but still, we would avoid to have too many assert lines

Sure let me do it.

Reply:
at least, test all the alphabets in simulation :)

Reply:
> Then what about the alphabets we don't have by name?

It's not callable by the main .py so would be fine with me. But if you want to test more, I'm always happy with it

Reply:
I just don't see how it would help. The implementation doesn't change between the alphabets, only the encoding. Not to mention if we want to test every alphabet, we need to create handpicked samples for each. Alternatively we can randomly sample, but we might miss extreme cases.

If you still insist, I can create another test to randomly sample from every alphabet and test like that. But we should keep this explicit test IMO.

Reply:
in tests, we should test all combination. Else, we may end up having stupid things like using "ACTG" instead of "ATCG" and not seing it. 

And making samples is not hard, if you use the vanilla implementation as the reference, which I do in my own tests. If you want to hardcode the expected values, you can call the vanilla implementation, see the result and keep it

Reply:
> we may end up having stupid things like using "ACTG" instead of "ATCG" and not seing it.

I didn't understand, sorry.

> And making samples is not hard, if you use the vanilla implementation as the reference, which I do in my own tests. If you want to hardcode the expected values, you can call the vanilla implementation, see the result and keep it

I included both approaches in tests.